codegen/model.py

```diff
@@ -823,6 +835,54 @@ def codegen(
         return outputs
 
 
+class MistralCode(VLlmDecoder):
+    def __init__(self, name: str, **kwargs) -> None:
+        kwargs["conversational"] = True
+        super().__init__(name, **kwargs)
+        self.eos += ["\n```"]
+
+    def codegen(
+        self, prompt: str, do_sample: bool = True, num_samples: int = 200
+    ) -> List[str]:
+        prompt = f"""Below is an instruction that describes a task. Write a response that appropriately completes the request.
+{prompt}
+"""
+        return VLlmDecoder.codegen(self, prompt, do_sample, num_samples)
+
+
+class Code(VLlmDecoder):
+    def __init__(self, name: str, **kwargs) -> None:
+        super().__init__(name, **kwargs)
+        self.eos += ["\n```"]
+
+    def codegen(
+        self, prompt: str, do_sample: bool = True, num_samples: int = 200
+    ) -> List[str]:
+        prompt = f"""This is a conversation with your helpful AI assistant. AI assistant can generate Python Code along with necessary explanation.
+
+Context
+You are a helpful AI assistant.
+
+USER: {prompt}
+ASSISTANT:"""

```

Comment:
Use a python block to wrap `{prompt}`.

Also for the ASSISTANT reply, end with "```python"